src/coreComponents/constitutive/fluid/PVTFunctions/BrineInternalEnergy.hpp

```diff
@@ -0,0 +1,160 @@
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file BrineInternalEnergy.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_BRINEINTERNALENERGY_HPP_
+#define GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_BRINEINTERNALENERGY_HPP_
+
+#include "PVTFunctionBase.hpp"
+
+#include "constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp"
+#include "functions/TableFunction.hpp"
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+namespace PVTProps
+{
+
+class BrineInternalEnergyUpdate final : public PVTFunctionBaseUpdate
+{
+public:
+
+  BrineInternalEnergyUpdate( arrayView1d< real64 const > const & componentMolarWeight )
+    : PVTFunctionBaseUpdate( componentMolarWeight )
+  {}
+
+  template< int USD1 >
+  GEOSX_HOST_DEVICE
+  void compute( real64 const & pressure,
+                real64 const & temperature,
+                arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                real64 & value,
+                bool useMass ) const;
+
+  template< int USD1, int USD2, int USD3, int USD4 >
+  GEOSX_HOST_DEVICE
+  void compute( real64 const & pressure,
+                real64 const & temperature,
+                arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                arraySlice1d< real64 const, USD2 > const & dPhaseComposition_dPressure,
+                arraySlice1d< real64 const, USD2 > const & dPhaseComposition_dTemperature,
+                arraySlice2d< real64 const, USD3 > const & dPhaseComposition_dGlobalCompFraction,
+                real64 & value,
+                real64 & dValue_dPressure,
+                real64 & dValue_dTemperature,
+                arraySlice1d< real64, USD4 > const & dValue_dGlobalCompFraction,
+                bool useMass ) const;
+
+  virtual void move( LvArray::MemorySpace const space, bool const touch ) override
+  {
+    PVTFunctionBaseUpdate::move( space, touch );
+  }
+
+protected:
+
+};
+
+class BrineInternalEnergy : public PVTFunctionBase
+{
+public:
+
+  BrineInternalEnergy( string const & name,
+                       string_array const & inputParams,
+                       string_array const & componentNames,
+                       array1d< real64 > const & componentMolarWeight )
+    : PVTFunctionBase( name,
+                       componentNames,
+                       componentMolarWeight )
+  {
+    // reserve for future: more accurate internal energy model should probably have some parameters
```

Comment:
I agree with you that at some point we should use a more accurate phase internal energy model, because right now it is quite simple :)